rpcs3/Emu/RSX/GL/GLGSRender.cpp

```diff
@@ -30,6 +30,25 @@ namespace
 		}
 		throw EXCEPTION("Unknow depth format");
 	}
+
+	u32 to_gl_internal_type(rsx::vertex_base_type type, u8 size)
+	{
+		/**
+		 * The buffer texture spec only allows fetches aligned to 8, 16, 32, etc...
+		 * This rules out most 3-component formats, except for the 32-wide RGB32F, RGB32I, RGB32UI
+		 */
+		const u32 vec1_types[] = { GL_R16, GL_R32F, GL_R16F, GL_R8, GL_R32I, GL_R16F, GL_R8 };
+		const u32 vec2_types[] = { GL_RG16, GL_RG32F, GL_RG16F, GL_RG8, GL_RG32I, GL_RG16F, GL_RG8 };
+		const u32 vec3_types[] = { GL_RGBA16, GL_RGB32F, GL_RGBA16F, GL_RGBA8, GL_RGB32I, GL_RGBA16F, GL_RGBA8 };	//VEC3 COMPONENTS NOT SUPPORTED!
+		const u32 vec4_types[] = { GL_RGBA16, GL_RGBA32F, GL_RGBA16F, GL_RGBA8, GL_RGBA32I, GL_RGBA16F, GL_RGBA8 };
+
+		const u32* vec_selectors[] = { 0, vec1_types, vec2_types, vec3_types, vec4_types };
+
+		if (type > rsx::vertex_base_type::ub256)
+			throw std::runtime_error(fmt::format("OpenGL error: unknown vertex base type 0x%X. file '%s' function '%s' line %ld", (u32)type, __FILE__, __FUNCTION__, __LINE__));
```

Comment:
std::runtime_error? Why not use the standard EXCEPTION, that we use in our codebase.
For example:
throw EXCEPTION("Unknown vertex base type: 0x%x", (u32)type);

It also logs the namespace, function and the line number already.